test/functional/ui/decorations_spec.lua

```diff
@@ -509,6 +509,98 @@ describe('decorations providers', function()
+          local win, buf, line = ...
+          api.nvim_buf_set_extmark(buf, test_ns, line, 0, {
+            virt_text = {{'+'}, {'1234567890', 'ErrorMsg'}};
+            virt_text_pos='eol_right_align';
+            ephemeral = true;
+          })
+        end
+      end
+    ]]
+
+    screen:expect{grid=[[
+      // just to see if there was an accident |
+      // on Mulholland Drive       +{2:1234567890}|
+      try_start();                 +{2:1234567890}|
+      bufref_T save_buf;           +{2:1234567890}|
+      switch_buffer(&save_buf, buf); +{2:12345678}|
+      posp = getmark(mark, false); +{2:1234567890}|
+      restore_buffer(&save_buf);^   +{2:1234567890}|
+                                              |
+    ]]}
+  end)
+
+  it('multiple instances of eol right align', function()
+    insert(mulholland)
+    setup_provider [[
+      local hl = api.nvim_get_hl_id_by_name "ErrorMsg"
+      local test_ns = api.nvim_create_namespace "mulholland"
+      function on_do(event, ...)
+        if event == "line" then
+          local win, buf, line = ...
+          api.nvim_buf_set_extmark(buf, test_ns, line, 0, {
+            virt_text = {{'11111'}};
+            virt_text_pos='eol_right_align';
+            ephemeral = true;
+          })
+          api.nvim_buf_set_extmark(0, test_ns, line, 0, {
+            virt_text = {{'22222'}};
+            virt_text_pos='eol_right_align';
+            ephemeral = true;
+          })
+        end
+      end
+    ]]
+
+    screen:expect{grid=[[
+      // just to see if there was an accident |
+      // on Mulholland Drive       11111 22222|
+      try_start();                 11111 22222|
+      bufref_T save_buf;           11111 22222|
+      switch_buffer(&save_buf, buf); 11111 222|
+      posp = getmark(mark, false); 11111 22222|
+      restore_buffer(&save_buf);^   11111 22222|
+                                              |
+    ]]}
+  end)
+
+  it('Right align prints virtual text on top of buffer text', function()
+    insert(mulholland)
+    setup_provider [[
+      local hl = api.nvim_get_hl_id_by_name "ErrorMsg"
+      local test_ns = api.nvim_create_namespace "mulholland"
+      function on_do(event, ...)
+        if event == "line" then
+          local win, buf, line = ...
+          api.nvim_buf_set_extmark(buf, test_ns, line, 0, {
+            virt_text = {{'+'}, {string.rep(' ', 5), 'ErrorMsg'}};
+            virt_text_pos='right_align';
+            ephemeral = true;
+          })
+        end
+      end
+    ]]
+
+    screen:expect{grid=[[
+      // just to see if there was an acc+{2:     }|
+      // on Mulholland Drive            +{2:     }|
+      try_start();                      +{2:     }|
+      bufref_T save_buf;                +{2:     }|
+      switch_buffer(&save_buf, buf);    +{2:     }|
+      posp = getmark(mark, false);      +{2:     }|
+      restore_buffer(&save_buf);^        +{2:     }|
+                                              |
+    ]]}
+  end)
+
```

Comment:
This test is not needed. It is nearly identical to the test `can have virtual text of the style: right_align`.

Reply:
You are correct; I've removed the test. I was attempting to more explicitly document the expected "write over buffer text" behavior through the test, but this benefit hardly outweighs the extra time it takes to run the test every time.